### PR TITLE
add include/exclude configuration

### LINKED ISSUE #4: Add whitelist/blacklist support
Currently, .launch file is created for all JavaExec tasks. Whitelist/blacklist support would be very useful especially when the default JavaExec task should be overridden with more sane defaults.

### PR DESCRIPTION
**Pull request by qinfchen**
fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +27,8 @@ The tasks are only added if their matching IDE plugin is applied.
 - `cleanEclipseLaunchConfig` - deletes the generated `.launch` files. Triggered when `:cleanEclipseProject` is called.
 - `ideaLaunchConfig` - generates the XML in the IDEA workspace. Triggered when `:ideaWorkspace` is called.
 
+Configuration
+-------------
```
Comment (jmcampanini) [inline README.md]: an empty section?

File: src/main/groovy/com/palantir/launchconfig/EclipseLaunchConfigTask.groovy [hunk 2]
```diff
@@ -66,4 +72,12 @@ class EclipseLaunchConfigTask extends DefaultTask {
         String xmlString = writer.toString() + "\n"
         Files.write(launchFile, xmlString.getBytes("UTF-8")).toFile()
     }
+
+    boolean isIncluded(String taskName) {
```
Comment (jmcampanini) [inline src/main/groovy/com/palantir/launchconfig/EclipseLaunchConfigTask.groovy]: can we rename to `shouldGenerate` or something more specific?

File: src/main/groovy/com/palantir/launchconfig/EclipseLaunchConfigTask.groovy [hunk 3]
```diff
@@ -15,16 +15,22 @@ class EclipseLaunchConfigTask extends DefaultTask {
...
     @TaskAction
     void generate() {
         this.project.tasks.withType(JavaExec) { javaExec ->
-            logger.debug("Generating launch file for '%s'.", javaExec.name)
-            writeLaunchFile(javaExec)
+            if (isIncluded(javaExec.name)) {
+	            logger.debug("Generating launch file for '%s'.", javaExec.name)
+	            writeLaunchFile(javaExec)
+			} else {
+                logger.debug("Skipping generation of '%s' since it is excluded in the launchConfig.", javaExec.name)
```
Comment (jmcampanini) [inline src/main/groovy/com/palantir/launchconfig/EclipseLaunchConfigTask.groovy]: there's also a bug here with the `%s`. check out issue #2 if you can fix that :-D

File: README.md (hunk 1 above)
Comment (qinfchen) [inline README.md]: oops

File: src/main/groovy/com/palantir/launchconfig/IdeaLaunchConfigTask.groovy [hunk 4]
```diff
@@ -59,4 +64,12 @@ class IdeaLaunchConfigTask extends DefaultTask {
     String determineRunConfigName(JavaExec javaExec) {
         return "${project.name}-${javaExec.name}"
     }
+
+    boolean isIncluded(String taskName) {
+        LaunchConfigExtension extension = this.project.extensions.getByType(LaunchConfigExtension)
+        Set<List> includedTasks = extension.getIncludedTasks()
+        Set<List> excludedTasks = extension.getExcludedTasks()
+
+        return ((includedTasks.isEmpty() || includedTasks.contains(taskName)) && !excludedTasks.contains(taskName));
```
Comment (jmcampanini) [inline src/main/groovy/com/palantir/launchconfig/IdeaLaunchConfigTask.groovy]: is the `isEmpty` check necessary? wouldn't the `.contains` call be sufficient?

File: src/main/groovy/com/palantir/launchconfig/LaunchConfigPlugin.groovy [hunk 5]
```diff
@@ -8,10 +8,14 @@ import org.gradle.plugins.ide.idea.IdeaPlugin
...
     public static final String GROUP_NAME = "Launch Config"
+    public static final String LAUNCH_EXTENSION = "launchConfig"
 
     @Override
     void apply(Project project) {
-        project.plugins.withType(EclipsePlugin) {
+
+        LaunchConfigExtension extension = project.extensions.create(LAUNCH_EXTENSION, LaunchConfigExtension)
+        
+		project.plugins.withType(EclipsePlugin) {
```
Comment (jmcampanini) [inline src/main/groovy/com/palantir/launchconfig/LaunchConfigPlugin.groovy]: weird indent?

File: src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy [hunk 6]
```diff
@@ -149,6 +149,124 @@ class EclipseLaunchConfigTaskIntegrationSpec extends IntegrationSpec {
...
+                main 'com.testing.HelloWorld'
+            }
+
+            task otherRun(type: JavaExec) {
+                classpath project.sourceSets.main.runtimeClasspath
+                main 'com.testing.HelloWorld'
+            }
+
+            launchConfig {
+				includedTasks = ['runDev']
```
Comment (jmcampanini) [inline src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy]: weird indent.

i would also prefer if this ended up being:

```groovy
launchConfig {
    includedTasks 'runDev', 'otherRun', 'etc'
}
```

also, thoughts on changing it to `include` and `exclude`?

File: src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy [hunk 7]
```diff
@@ -183,7 +301,6 @@ class EclipseLaunchConfigTaskIntegrationSpec extends IntegrationSpec {
     def "cleanEclipse should delete the file"() {
         setup:
         writeHelloWorld("com.testing")
-        createFile('runDev.launch') << "original content"
```
Comment (jmcampanini) [inline src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy]: why is this deleted?

**Comment by jmcampanini** [comment]
seems reasonable. i'm not 100% on board with the DSL for the configuration block, but i like the functionality. see comments inline re: DSL

File: src/main/groovy/com/palantir/launchconfig/IdeaLaunchConfigTask.groovy (hunk 4 above)
Comment (qinfchen) [inline src/main/groovy/com/palantir/launchconfig/IdeaLaunchConfigTask.groovy]: yes, it is. empty means all tasks are included. Will add comment

File: src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy (hunk 6 above)
Comment (qinfchen) [inline src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy]: will change

File: src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy (hunk 7 above)
Comment (qinfchen) [inline src/test/groovy/com/palantir/launchconfig/EclipseLaunchConfigTaskIntegrationSpec.groovy]: the test is not testing this file. copy paste mistake :)
